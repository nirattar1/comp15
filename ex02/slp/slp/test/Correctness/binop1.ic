class My_Class {
	void doNothing(){
		int x = 1 - 2 * 6 + 3;
		
		int y = 4 + 6 / 6 + 3;

		int y = 3 % 4 + 3;   	

		boolean x = (1 < 3) || (4 > 5) ;
	}
	
}