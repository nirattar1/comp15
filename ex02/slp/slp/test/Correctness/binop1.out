Parsed C:\Users\Vadim\Documents\Studies\Compilation\binop1.ic successfully!

Abstract Syntax Tree: C:\Users\Vadim\Documents\Studies\Compilation\binop1.ic

1: Declaration of class: My_Class
  2: Declaration of virtual method: doNothing
    2: Primitive data type: void
    3: Declaration of local variable: x, with initial value
      3: Primitive data type: int
      3: Mathematical binary operation: addition
        3: Mathematical binary operation: subtraction
          3: Integer literal: 1
          3: Mathematical binary operation: multiplication
            3: Integer literal: 2
            3: Integer literal: 6
        3: Integer literal: 3
    5: Declaration of local variable: y, with initial value
      5: Primitive data type: int
      5: Mathematical binary operation: addition
        5: Mathematical binary operation: addition
          5: Integer literal: 4
          5: Mathematical binary operation: division
            5: Integer literal: 6
            5: Integer literal: 6
        5: Integer literal: 3
    7: Declaration of local variable: y, with initial value
      7: Primitive data type: int
      7: Mathematical binary operation: addition
        7: Mathematical binary operation: modulo
          7: Integer literal: 3
          7: Integer literal: 4
        7: Integer literal: 3
    9: Declaration of local variable: x, with initial value
      9: Primitive data type: boolean
      9: Logical binary operation: logical or
        9: Logical binary operation: less than
          9: Integer literal: 1
          9: Integer literal: 3
        9: Logical binary operation: greater than
          9: Integer literal: 4
          9: Integer literal: 5
